crates/project/src/project.rs

```diff
@@ -7708,15 +7709,29 @@ impl Project {
                 anyhow::Ok((repo, relative_path, content))
             });
 
-            cx.background_executor().spawn(async move {
-                let (repo, relative_path, content) = blame_params?;
-                let lock = repo.lock();
-                lock.blame(&relative_path, content)
+            cx.to_async().spawn(|mut cx| {
+                let this = this.clone();
+                async move {
+                    let (repo, relative_path, content) = blame_params?;
+                    let lock = repo.lock();
+                    let blame_res = lock
+                        .blame(&relative_path, content)
+                        .with_context(|| format!("Failed to git blame {relative_path:?}"));
+
+                    if let Err(ref err) = blame_res {
```

Comment:
I'm not @mrnugget , but I guess forcing this to be on a foreground thread is a pretty big deal; wouldn't it work if we emitted a notification on error in the caller of `blame_buffer`?

Reply:
yeah, good point. did exactly that.